src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1059,6 +1075,27 @@ internal void DeleteModelInternal(List<ModelBase> modelsToDelete)
             OnDeletionComplete(this, EventArgs.Empty);
         }
 
+        internal void UngroupModel(List<ModelBase> modelsToUngroup)
```

Comment:
This is same as DeleteModelInternal, with a difference. Node and the annotation model is not deleted. They are just uncoupled here.